user/models.py

```diff
@@ -48,24 +37,35 @@ def role(self):
         else:
             return "Contributor"
 
+    # We write a setter as we construct temporary users when doing password validation, and sometimes role information
+    # is part of the data.
     @role.setter
     def role(self, value):
-        self.__roles_cache = set(value or self.groups.values_list("name", flat=True))
-
-    def get_all_roles(self):
-        if self.__roles_cache is None:
-            self.role = None  # force cache to update
-        return self.__roles_cache
+        pass
 
     def at_least(self, role):
-        return role in self.get_all_roles()
+        """Returns if a user has at least a given role's privileges"""
+        return self.is_staff or role == self.role or role in GROUPS.get(self.role, [])
+
+    @property
+    def is_editor(self):
```

Comment:
Don't think we need to explicitly deprecate this since it's basically only the two of us controlling the codebase and we can just remove any usage.

Reply:
I wanted to write in a deprecation warning so we're aware we should remove it, since there's a non-negative chance I'll forget to do so.